source/dotnet/Library/AdaptiveCards/AdaptiveTypedElementConverter.cs

```diff
@@ -66,84 +55,115 @@ public static void RegisterTypedElement<T>(string typeName = null)
+                if (objectId == null)
                 {
                     if (typeof(AdaptiveInput).GetTypeInfo().IsAssignableFrom(type.GetTypeInfo()))
                     {
                         throw new AdaptiveSerializationException($"Required property 'id' not found on '{typeName}'");
                     }
                 }
-                else
+
+                // add id of element to ParseContext
+                AdaptiveInternalID internalID = AdaptiveInternalID.Current();
+                if (type != typeof(AdaptiveCard))
                 {
-                    string objectId = jObject.Value<string>("id");
-                    if (Ids.Contains(objectId))
-                    {
-                        throw new AdaptiveSerializationException($"Duplicate 'id' found: '{objectId}'");
-                    }
-                    else
-                    {
-                        Ids.Add(objectId);
-                    }
+                    internalID = AdaptiveInternalID.Next();
+                    ParseContext.PushElement(objectId, internalID);
                 }
 
                 var result = (AdaptiveTypedElement)Activator.CreateInstance(type);
                 try
                 {
                     serializer.Populate(jObject.CreateReader(), result);
                 }
-                catch (JsonSerializationException)
+                catch (JsonSerializationException) { }
+
+                // remove id of element from ParseContext
+                if (type != typeof(AdaptiveCard))
+                    ParseContext.PopElement();
+
+                HandleAdditionalProperties(result);
+                return result;
+            }
+            else // We're looking at an unknown element
+            {
+                string objectId = jObject.Value<string>("id");
+                AdaptiveInternalID internalID = AdaptiveInternalID.Next();
+
+                // Handle deserializing unknown element
+                ParseContext.PushElement(objectId, internalID);
+                AdaptiveTypedElement result = null;
+                if (ParseContext.Type == ParseContext.ContextType.Element)
+                {
+                    try
+                    {
+                        result = (AdaptiveTypedElement)Activator.CreateInstance(typeof(AdaptiveUnknownElement));
+                        serializer.Populate(jObject.CreateReader(), result);
+                    }
+                    catch (JsonSerializationException) { }
+                }
+                else // ParseContext.Type == ParseContext.ContextType.Action
                 {
-                    return result;
+                    try
+                    {
+                        result = (AdaptiveTypedElement)Activator.CreateInstance(typeof(AdaptiveUnknownAction));
+                        serializer.Populate(jObject.CreateReader(), result);
+                    }
+                    catch (JsonSerializationException) { }
```

Comment:
>   catch (JsonSerializationException) { } [](start = 17, length = 41)

Why are we just swallowing exceptions here?

Reply:
actually, these aren't needed :)

---
In reply to: [278323956](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278323956) [](ancestors = 278323956)